demisto_sdk/commands/pre_commit/hooks/docker.py

```diff
@@ -0,0 +1,305 @@
+    """
+    Adds the native image images into the dict with the files that should be run on them
+    Args:
+        tags_to_files: Dict[str, Tuple[str, dict] the incoming dict without native image of files split according to the docker images
+        docker_image_flag: the flag from the config file. all/native:ga/native:maintenance etc
+
+    Returns: The updated dict with the native images.
+
+    """
+    docker_flags = set(docker_image_flag.split(","))
+    all_tags_to_files = defaultdict(list)
+    native_image_config = NativeImageConfig.get_instance()
+
+    for image, scripts in tags_to_files.items():
+        for file, yml in scripts:
+
+            supported_native_images = ScriptIntegrationSupportedNativeImages(
+                _id=yml.get("commonfields", {}).get("id", ""),
+                native_image_config=native_image_config,
+                docker_image=image,
+            ).get_supported_native_docker_tags(docker_flags)
+            for native_image in supported_native_images:
+                all_tags_to_files[native_image].append((file, yml))
+            if {
+                DockerImageFlagOption.FROM_YML.value,
+                DockerImageFlagOption.ALL_IMAGES.value,
+            } & docker_flags:
+                all_tags_to_files[image].append((file, yml))
+    return all_tags_to_files
+
+
+@functools.lru_cache
+def get_yml_for_code(code_file) -> Optional[dict]:
+    yml_in_directory = [f for f in os.listdir(code_file.parent) if f.endswith(".yml")]
+    if (
+        len(yml_in_directory) == 1
+        and (yml_file := code_file.parent / yml_in_directory[0]).is_file()
+    ):
+        try:
+            return get_yaml(yml_file)
+        except Exception:
+            logger.debug(f"Could not parse file {code_file}")
+    return None
+    # could be reasonable cant parse. We have some non-parsable ymls for tests
+
+
+def docker_tag_to_runfiles(
+    files_to_run: Iterable, docker_image_flag
+) -> Dict[str, List[Tuple[str, Dict]]]:
+    """
+    Iterates over all files and finds the files assosciated yml. Groups the files by the dockerimages
+    Args:
+        files_to_run: PosixFiles to run the command on
+        docker_image_flag: the docker_image config value
+
+    Returns: A dict of image to List of files(Tuple[path, yml]) including native images
+
+    """
+    tags_to_files = defaultdict(list)
+    for file in files_to_run:
+        yml: Optional[dict] = get_yml_for_code(file)
+        if not yml:
+            continue
+        for docker_image in docker_images_for_file(yml):
+            tags_to_files[docker_image].append((file, yml))
+    return with_native_tags(tags_to_files, docker_image_flag)
+
+
+def docker_images_for_file(yml: dict) -> set:
+    """
+    Args:
+        yml: the yml representation of the content item
+
+    Returns: all docker images (without native) that a file should tested on
+
+    """
+    images_to_return = set()
+    if image := yml.get("dockerimage"):
```

Comment:
you can use integration_script.docker_image

Regarding alt_dockerimages: we need to add this property to the object.